### PR TITLE
Add new plan for MDA point distribution

### PR DESCRIPTION
**Pull request by ciremusyoka**
Addresses https://github.com/OpenSRP/opensrp-client-reveal/issues/677 and https://github.com/OpenSRP/opensrp-client-reveal/issues/730

- Add MDA point intervention type
- Add MDA point activities
- Add ability to update Jurisdiction API end point based on selected intervention type

### CODE REVIEW

File: src/components/forms/JurisdictionSelect/index.tsx [hunk 1]
```diff
@@ -53,13 +56,41 @@ const defaultProps: Partial<JurisdictionSelectProps> = {
...
+    setCloseMenuOnSelect(false);
+    // set the Formik field value
+    if (form && field) {
+      form.setFieldValue(field.name, '');
+    }
+  };
+
+  useEffect(() => {
+    if (interventionType === InterventionType.MDAPoint) {
+      resetStateVars();
```
Comment (moshthepitt) [inline src/components/forms/JurisdictionSelect/index.tsx:90]: @ciremusyoka why do we need to do this?

File: src/components/forms/PlanForm/helpers.ts [hunk 2]
```diff
@@ -44,8 +44,9 @@ import {
 } from './types';
 
 /** separate FI and IRS activities */
```
Comment (moshthepitt) [inline src/components/forms/PlanForm/helpers.ts:46]: Should this comment be updated

File: src/components/forms/PlanForm/index.tsx [hunk 3]
```diff
@@ -121,6 +126,10 @@ export type LocationChildRenderProp = (
   index: number
 ) => JSX.Element;
 
+// /** intervention types */
+// const { FI, IRS, MDA, MDAPoint } = InterventionType;
+
+// const currentInterventionType: typeof FI | typeof IRS | typeof MDA | typeof MDAPoint = InterventionType.FI;
```
Comment (moshthepitt) [inline src/components/forms/PlanForm/index.tsx:132]: @ciremusyoka why comment this out?

File: src/components/forms/PlanForm/index.tsx [hunk 4]
```diff
@@ -162,6 +174,24 @@ const PlanForm = (props: PlanFormProps) => {
   const editMode: boolean = initialValues.identifier !== '';
   const irsActivities = getFormActivities(IRSActivities);
   const fiActivities = getFormActivities(FIActivities);
+  const mdaPointActivities = getFormActivities(MDAPointActivities);
+
+  if (selectedIntervention === InterventionType.MDAPoint) {
+    const jurisdictionSelectProp: Partial<JurisdictionSelectProps> = {
+      apiEndpoint: 'location/findByProperties',
```
Comment (moshthepitt) [inline src/components/forms/PlanForm/index.tsx:181]: Lets move this to the constants module

File: src/components/forms/PlanForm/tests/fixtures.ts [hunk 5]
```diff
@@ -629,6 +703,32 @@ export const extractedActivitiesFromForms = [
     timingPeriodEnd: timingPeriodEnd.toISOString(),
     timingPeriodStart: timingPeriodStart.toISOString(),
   },
+  {
+    actionCode: 'MDA Adverse Event(s)',
+    actionDescription: 'Report any adverse events from medication',
+    actionIdentifier: '',
```
Comment (moshthepitt) [inline src/components/forms/PlanForm/tests/fixtures.ts:709]: Do we need the actionIdentifier to be blank?

File: src/configs/settings.ts [hunk 6]
```diff
@@ -307,7 +317,7 @@ export interface PlanAction {
 
 /** Plan Goal detailQuantity */
 export interface PlanGoaldetailQuantity {
-  comparator: '>=';
+  comparator: '>=' | '<=';
```
Comment (moshthepitt) [inline src/configs/settings.ts:320]: @ciremusyoka why are we adding `'<='`?

Comment (ciremusyoka) [reply src/configs/settings.ts:320]: @moshthepitt this is based on [this file](https://docs.google.com/spreadsheets/d/1kGPgyF0eJdag19FFL1Oog2gdphdPtKobY4EQ0gMgtgw/edit#gid=1805975071) line 262.

Comment (moshthepitt) [reply src/configs/settings.ts:320]: Cool, thanks

File: src/components/forms/PlanForm/tests/fixtures.ts (hunk 5 above)
Comment (ciremusyoka) [reply src/components/forms/PlanForm/tests/fixtures.ts:709]: My thought is `actionIdentifier` comes from the API. Even when defining the actions on [settings](https://github.com/onaio/reveal-frontend/blob/14bfb21d52016673392fc6e18a7d5d5b33ac37f5/src/configs/settings.ts#L354) it is left blank

File: src/components/forms/JurisdictionSelect/index.tsx (hunk 1 above)
Comment (ciremusyoka) [reply src/components/forms/JurisdictionSelect/index.tsx:90]: @moshthepitt This is meant to make the code for resetting the component state and form more reusable. 
Needed to use the function [here](https://github.com/onaio/reveal-frontend/blob/814ad39c716c2990a0f8d1e69df3c0783db91b3a/src/components/forms/JurisdictionSelect/index.tsx#L88-L92) too.

File: src/components/forms/PlanForm/index.tsx (hunk 3 above)
Comment (ciremusyoka) [reply src/components/forms/PlanForm/index.tsx]: Have removed it.

File: src/components/forms/JurisdictionSelect/index.tsx (hunk 1 above)
Comment (moshthepitt) [reply src/components/forms/JurisdictionSelect/index.tsx:90]: @ciremusyoka thanks.  However, why do we need to call `resetStateVars()` when the component mounts and `interventionType === InterventionType.MDAPoint`?

Comment (ciremusyoka) [reply src/components/forms/JurisdictionSelect/index.tsx:90]: @moshthepitt Will `IRS, FI and MDAPoint` intervention types use similar endpoint to pull locations?

Comment (moshthepitt) [reply src/components/forms/JurisdictionSelect/index.tsx:90]: Yes

**Comment by moshthepitt** [comment]
@ciremusyoka  can you add a screenshot(s) please?

**Comment by ciremusyoka** [comment]
UI screen shots
point-1
point2
point-3

File: src/components/forms/PlanForm/index.tsx [hunk 7]
```diff
@@ -705,43 +717,46 @@ const PlanForm = (props: PlanFormProps) => {
                               value={values.activities[index].actionDescription}
                             />
                           </FormGroup>
-                          <FormGroup>
-                            <Label for={`activities-${index}-goalValue`}>{QUANTITY_LABEL}</Label>
-                            <InputGroup id={`activities-${index}-goalValue-input-group`}>
-                              <Field
-                                type="number"
+                          {values.activities[index].actionCode !==
```
Comment (moshthepitt) [inline src/components/forms/PlanForm/index.tsx:720]: Why are we hiding this field for MDAPoint?

**Comment by moshthepitt** [comment]
When loading this page http://localhost:3000/plans/update/6cec8ae0-8aa6-410f-8d0d-532a5e5fffb5

I get this error:

```
index.js:1437 Warning: A component is changing a controlled input of type text to be uncontrolled. Input elements should not switch from controlled to uncontrolled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components
    in input (created by Input)
    in Input (at Search/index.tsx:56)
    in div (created by FormGroup)
    in FormGroup (at Search/index.tsx:55)
    in form (created by Form)
    in Form (at Search/index.tsx:54)
    in SearchForm (at PlanDefinitionList/index.tsx:127)
    in div (at PlanDefinitionList/index.tsx:110)
    in PlanDefinitionList (created by Context.Consumer)
    in Connect(PlanDefinitionList) (created by Route)
    in Route (created by PrivateRoute)
    in PrivateRoute (created by Context.Consumer)
    in Connect(PrivateRoute) (at App.tsx:148)
    in Switch (at App.tsx:146)
    in div (created by Col)
    in Col (at App.tsx:144)
    in div (created by Row)
    in Row (at App.tsx:143)
    in div (created by Container)
    in Container (at App.tsx:142)
    in div (created by Container)
    in Container (at App.tsx:139)
    in App (at src/index.tsx:19)
    in Router (created by ConnectedRouter)
    in ConnectedRouter (created by Context.Consumer)
    in ConnectedRouterWithContext (created by Context.Consumer)
    in Connect(ConnectedRouterWithContext) (at src/index.tsx:18)
    in Provider (at src/index.tsx:17)
console.<computed> @ index.js:1437
r @ react_devtools_backend.js:6
printWarning @ react-dom.development.js:89
error @ react-dom.development.js:61
updateWrapper @ react-dom.development.js:1889
updateProperties @ react-dom.development.js:6266
commitUpdate @ react-dom.development.js:7569
commitWork @ react-dom.development.js:20405
commitMutationEffects @ react-dom.development.js:22635
callCallback @ react-dom.development.js:189
invokeGuardedCallbackDev @ react-dom.development.js:238
invokeGuardedCallback @ react-dom.development.js:291
commitRootImpl @ react-dom.development.js:22372
unstable_runWithPriority @ scheduler.development.js:659
runWithPriority$1 @ react-dom.development.js:11077
commitRoot @ react-dom.development.js:22246
finishSyncRender @ react-dom.development.js:21663
performSyncWorkOnRoot @ react-dom.development.js:21649
(anonymous) @ react-dom.development.js:11131
unstable_runWithPriority @ scheduler.development.js:659
runWithPriority$1 @ react-dom.development.js:11077
flushSyncCallbackQueueImpl @ react-dom.development.js:11126
flushSyncCallbackQueue @ react-dom.development.js:11114
flushPassiveEffectsImpl @ react-dom.development.js:22738
unstable_runWithPriority @ scheduler.development.js:659
runWithPriority$1 @ react-dom.development.js:11077
flushPassiveEffects @ react-dom.development.js:22679
(anonymous) @ react-dom.development.js:22558
workLoop @ scheduler.development.js:603
flushWork @ scheduler.development.js:559
performWorkUntilDeadline @ scheduler.development.js:162
```

Is it related?  Can you look at it and fix if it is, otherwise create an issue?

File: src/components/forms/PlanForm/index.tsx (hunk 7 above)
Comment (moshthepitt) [reply src/components/forms/PlanForm/index.tsx:720]: Actually @ciremusyoka for MDA Adverse Event(s) I am worried that completely removing the field makes things inconsistent.  Can you instead show it, but make it disabled? (i.e. not user editable)

**Comment by moshthepitt** [comment]
@gstuder-ona this is the sample JSON for a plan generated by this form:

```json
{
    "action": [
        {
            "code": "MDA Adverse Event(s)",
            "description": "Report any adverse events from medication",
            "goalId": "Point_adverse_effect_MDA",
            "identifier": "d0988a3d-fb02-51c6-a005-50ad79a03db7",
            "prefix": 1,
            "reason": "Routine",
            "subjectCodableConcept": {
                "text": "MDA_Point_Adverse_Event"
            },
            "taskTemplate": "MDA_Point_Adverse_Event",
            "timingPeriod": {
                "end": "2020-05-22",
                "start": "2020-05-15"
            },
            "title": "MDA Adverse Event(s)"
        },
        {
            "code": "MDA Dispense",
            "description": "Dispense medication to each eligible person",
            "goalId": "Point_dispense_MDA",
            "identifier": "410482dd-1b06-5f87-b069-5ab3ea5caa75",
            "prefix": 2,
            "reason": "Routine",
            "subjectCodableConcept": {
                "text": "MDA_Point_Dispense"
            },
            "taskTemplate": "MDA_Point_Dispense",
            "timingPeriod": {
                "end": "2020-05-22",
                "start": "2020-05-15"
            },
            "title": "MDA Dispense"
        }
    ],
    "goal": [
        {
            "description": "Report any adverse events from medication",
            "id": "Point_adverse_effect_MDA",
            "priority": "medium-priority",
            "target": [
                {
                    "detail": {
                        "detailQuantity": {
                            "comparator": "<=",
                            "unit": "Percent",
                            "value": 2
                        }
                    },
                    "due": "2020-05-22",
                    "measure": "Percent of people who reported adverse events"
                }
            ]
        },
        {
            "description": "Dispense medication to each eligible person",
            "id": "Point_dispense_MDA",
            "priority": "medium-priority",
            "target": [
                {
                    "detail": {
                        "detailQuantity": {
                            "comparator": ">=",
                            "unit": "Percent",
                            "value": 100
                        }
                    },
                    "due": "2020-05-22",
                    "measure": "Percent of eligible people"
                }
            ]
        }
    ],
    "date": "2020-05-15",
    "effectivePeriod": {
        "end": "2020-06-04",
        "start": "2020-05-15"
    },
    "identifier": "276c7f2c-c2a7-5478-a005-8ef4ac753e99",
    "jurisdiction": [
        {
            "code": "3953"
        }
    ],
    "name": "MDA-Point-2020-05-15",
    "status": "draft",
    "title": "MDA-Point 2020-05-15 test by mosh",
    "useContext": [
        {
            "code": "interventionType",
            "valueCodableConcept": "MDA-Point"
        },
        {
            "code": "taskGenerationStatus",
            "valueCodableConcept": "False"
        }
    ],
    "version": 2
}
```

**Comment by ciremusyoka** [comment]
> When loading this page http://localhost:3000/plans/update/6cec8ae0-8aa6-410f-8d0d-532a5e5fffb5
> 
> I get this error:
> 
> ```
> index.js:1437 Warning: A component is changing a controlled input of type text to be uncontrolled. Input elements should not switch from controlled to uncontrolled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components
>     in input (created by Input)
>     in Input (at Search/index.tsx:56)
>     in div (created by FormGroup)
>     in FormGroup (at Search/index.tsx:55)
>     in form (created by Form)
>     in Form (at Search/index.tsx:54)
>     in SearchForm (at PlanDefinitionList/index.tsx:127)
>     in div (at PlanDefinitionList/index.tsx:110)
>     in PlanDefinitionList (created by Context.Consumer)
>     in Connect(PlanDefinitionList) (created by Route)
>     in Route (created by PrivateRoute)
>     in PrivateRoute (created by Context.Consumer)
>     in Connect(PrivateRoute) (at App.tsx:148)
>     in Switch (at App.tsx:146)
>     in div (created by Col)
>     in Col (at App.tsx:144)
>     in div (created by Row)
>     in Row (at App.tsx:143)
>     in div (created by Container)
>     in Container (at App.tsx:142)
>     in div (created by Container)
>     in Container (at App.tsx:139)
>     in App (at src/index.tsx:19)
>     in Router (created by ConnectedRouter)
>     in ConnectedRouter (created by Context.Consumer)
>     in ConnectedRouterWithContext (created by Context.Consumer)
>     in Connect(ConnectedRouterWithContext) (at src/index.tsx:18)
>     in Provider (at src/index.tsx:17)
> console.<computed> @ index.js:1437
> r @ react_devtools_backend.js:6
> printWarning @ react-dom.development.js:89
> error @ react-dom.development.js:61
> updateWrapper @ react-dom.development.js:1889
> updateProperties @ react-dom.development.js:6266
> commitUpdate @ react-dom.development.js:7569
> commitWork @ react-dom.development.js:20405
> commitMutationEffects @ react-dom.development.js:22635
> callCallback @ react-dom.development.js:189
> invokeGuardedCallbackDev @ react-dom.development.js:238
> invokeGuardedCallback @ react-dom.development.js:291
> commitRootImpl @ react-dom.development.js:22372
> unstable_runWithPriority @ scheduler.development.js:659
> runWithPriority$1 @ react-dom.development.js:11077
> commitRoot @ react-dom.development.js:22246
> finishSyncRender @ react-dom.development.js:21663
> performSyncWorkOnRoot @ react-dom.development.js:21649
> (anonymous) @ react-dom.development.js:11131
> unstable_runWithPriority @ scheduler.development.js:659
> runWithPriority$1 @ react-dom.development.js:11077
> flushSyncCallbackQueueImpl @ react-dom.development.js:11126
> flushSyncCallbackQueue @ react-dom.development.js:11114
> flushPassiveEffectsImpl @ react-dom.development.js:22738
> unstable_runWithPriority @ scheduler.development.js:659
> runWithPriority$1 @ react-dom.development.js:11077
> flushPassiveEffects @ react-dom.development.js:22679
> (anonymous) @ react-dom.development.js:22558
> workLoop @ scheduler.development.js:603
> flushWork @ scheduler.development.js:559
> performWorkUntilDeadline @ scheduler.development.js:162
> ```
> 
> Is it related? Can you look at it and fix if it is, otherwise create an issue?

@moshthepitt i don't think it is related but will have a look on it

**Comment by ciremusyoka** [comment]
> When loading this page http://localhost:3000/plans/update/6cec8ae0-8aa6-410f-8d0d-532a5e5fffb5
> 
> I get this error:
> 
> ```
> index.js:1437 Warning: A component is changing a controlled input of type text to be uncontrolled. Input elements should not switch from controlled to uncontrolled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components ........
> ```
> 
> Is it related? Can you look at it and fix if it is, otherwise create an issue?

This error was caused by Search form. 
At some point `undefined` was being passed as the input value

**Comment by moshthepitt** [comment]
> This error was caused by Search form.
At some point undefined was being passed as the input value

@ciremusyoka could you please create an issue?  cc: @p-netm

**Comment by ciremusyoka** [comment]
> > This error was caused by Search form.
> > At some point undefined was being passed as the input value
> 
> @ciremusyoka could you please create an issue? cc: @p-netm

@moshthepitt have fixed it.

**Pull request merged**